modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp

```diff
@@ -620,11 +620,16 @@ void GenericConstraintProblem::clear(int nbC)
 
 void GenericConstraintProblem::freeConstraintResolutions()
 {
-    for(auto* constraintsResolution : constraintsResolutions)
+    for(unsigned int i=0; i<constraintsResolutions.size(); i++)
     {
-        delete constraintsResolution;
+        if (constraintsResolutions[i] != nullptr)
+        {
+            delete constraintsResolutions[i];
+            constraintsResolutions[i] = nullptr;
+        }
     }
 }
```

Comment:
Maybe if we want to keep the modern for-loop, the fix would be like this: (cannot make a suggestion)
```
    for(auto* constraintsResolution : constraintsResolutions)
    {
        if (constraintsResolution != nullptr)
        {
            delete constraintsResolution;
            constraintsResolution = nullptr;
        }
    }
```

Reply:
I am not sure it would do it. If I am not wrong, `constraintsResolution` is a copy of a pointer, and not the actual element of the vector. See my suggestion in my review.

Reply:
Also, `nullptr` is not necessary. Deallocation of a null pointer does nothing (see https://en.cppreference.com/w/cpp/memory/new/operator_delete)

Reply:
> 
> 
> I am not sure it would do it. If I am not wrong, `constraintsResolution` is a copy of a pointer, and not the actual element of the vector. See my suggestion in my review.

nice catch 👍